### PR TITLE
build: warn the developer if BUILD_DRY_RUN is true

### PR DESCRIPTION
**Pull request by joejulian**
... also linter fix

**What problem does this PR solve?**:
Warn about a hidden undocumented environment variable.

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by faiq — APPROVED** [review]

**Comment by dlipovetsky** [comment]
Seems odd to have a warning about the default value. Maybe we were wrong to make it the default?

Or maybe the problem is that we're using make as a CLI framework, but only env vars, and no flags? :laughing:

**Comment by joejulian** [comment]
Maybe so. I don't know why it exists and why it has that default. I ran a make command twice and couldn't see why it wasn't creating an ami. I asked @faiq if it worked for him, and he told me about this variable. I'm trying to add this warning so that whatever reason it exists, any new developers that don't already know that secret will at least have some way of learning about it.

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,9 @@ INTERACTIVE := $(shell [ -t 0 ] && echo 1)
 
 # BUILD_DRY_RUN determines the value of the --dry-run flag of the build command. Should be 'true' or 'false'.
```
Comment (dlipovetsky) [inline Makefile:8]:
```suggestion
# BUILD_DRY_RUN enables the build command's dry-run mode. In this mode, the build does not produce any permanent artifacts, e.g., an AMI. Should be 'true' or 'false'.
```

**Comment by dlipovetsky** [comment]
> Maybe so. I don't know why it exists and why it has that default. I ran a make command twice and couldn't see why it wasn't creating an ami. I asked @faiq if it worked for him, and he told me about this variable. I'm trying to add this warning so that whatever reason it exists, any new developers that don't already know that secret will at least have some way of learning about it.

Which make target did you run? Maybe that target should disable dry-run?

We added dry-run (and made it the default) in order to avoid creating permanent artifacts as the result of running e2e tests of the build command.

File: Makefile (hunk 1 above)
Comment (joejulian) [reply Makefile:8]: It's not about the comment, it's that I wouldn't have even guessed that I would have to read the Makefile to understand this variable even exists. With this warning, it gives that hint.

**Comment by joejulian** [comment]
> > Maybe so. I don't know why it exists and why it has that default. I ran a make command twice and couldn't see why it wasn't creating an ami. I asked @faiq if it worked for him, and he told me about this variable. I'm trying to add this warning so that whatever reason it exists, any new developers that don't already know that secret will at least have some way of learning about it.
> 
> Which make target did you run? Maybe that target should disable dry-run?
> 
> We added dry-run (and made it the default) in order to avoid creating permanent artifacts as the result of running e2e tests of the build command.

That makes total sense and for that reason, I think the default should remain. As a developer, it's probably more common that you'll want to test the build process without creating the ami artifact.

File: Makefile (hunk 1 above)
Comment (dlipovetsky) [reply Makefile:8]: Good point. Would you like to move my text here to the warning you added?

**Review by dlipovetsky — APPROVED** [review]

**Pull request merged**